core/src/main/java/com/linecorp/armeria/server/ServerPort.java

```diff
@@ -133,28 +131,42 @@ public boolean hasTls() {
      * {@link SessionProtocol#H2C} is in the list of {@link SessionProtocol}s.
      */
     public boolean hasHttp() {
-        return hasProtocol(HTTP) || hasProtocol(H1C) || hasProtocol(H2C);
+        return httpValues().stream().anyMatch(this::hasExactProtocol);
```

Comment:
How about just:
```suggestion
        return hasExactProtocol(HTTP);
```
because `protocols` can contain only `HTTP`, `HTTPS`, and `PROXY`?